shapely/constructive.py

```diff
@@ -1429,6 +1430,44 @@ def minimum_bounding_circle(geometry, **kwargs):
     return lib.minimum_bounding_circle(geometry, **kwargs)
 
 
+@multithreading_enabled
+def minimum_width(geometry, **kwargs):
+    """Compute the minimum width (minimum diameter) of a geometry.
+
+    The minimum width is defined as the width of the smallest band that
+    contains the geometry, where a band is a strip of the plane defined
+    by two parallel lines. This is also known as the minimum diameter.
+
+    For most geometry types, this function returns a LineString representing
+    the minimum width line segment. For degenerate cases (points), it may
+    return the point itself.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+        Geometry or geometries for which to compute the minimum width.
+    **kwargs
+        See :ref:`NumPy ufunc docs <ufuncs.kwargs>` for other keyword arguments.
+
+    Examples
+    --------
+    >>> import shapely
+    >>> from shapely import Point, LineString, Polygon
+    >>> polygon = Polygon([(0, 0), (1, 0), (1, 1), (0, 1), (0, 0)])
+    >>> shapely.minimum_width(polygon)  # doctest: +SKIP
```

Comment:
This skips should not be needed?

```suggestion
    >>> shapely.minimum_width(polygon)
```